pkg/api/v1alpha1/awsiamconfig.go

```diff
@@ -98,3 +98,28 @@ func validateMapUsers(mapUsers []MapUsers) error {
 	}
 	return nil
 }
+
+func validateAWSIamRefName(config *AWSIamConfig, refName string) error {
+	if config == nil {
+		return nil
+	}
+
+	if config.Name != refName {
+		return fmt.Errorf("AWSIamConfig retrieved with name %s does not match name (%s) specified in "+
+			"identityProviderRefs", config.Name, refName)
+	}
+
+	return nil
+}
+
+func validateAWSIamNamespace(config *AWSIamConfig, clusterConfig *Cluster) error {
```

Comment:
this only reason this function exists is to keep current validation logic working? eventually we dont need this right? since it's taken care of in cluster.validation

Reply:
Yeah
My plan was to deprecate all the "parsing" methods in this package once we start using the `cluster.Config` in the `cluster.Spec`
And once that's done, the RefName and Namespace validations won't be necessary in this package